### DISCUSSION TITLE
Add support to get remaining TTL

### DISCUSSION THREAD

**Discussion by surajpaypay** [redis/lettuce#2918] [Q&A] [answered]
## Feature Request
I want to use the get TL of key  but there is no support for it. Please let me know if there is any work around. My implementation class is RedisStringCommands.

#### Is your feature request related to a problem? Please describe

https://redis.io/docs/latest/commands/ttl/

#### Describe the solution you'd like
public int getTtl(String key);

**Comment by tishun** [comment, marked as answer]
Hey @surajpaypay ,

is the [RedisKeyCommands.ttl(K key)](https://lettuce.io/core/release/api/io/lettuce/core/api/sync/RedisKeyCommands.html#ttl-K-) what you are searching for?

> My implementation class is RedisStringCommands.

Not sure what you mean by that, could you clarify?

**Reply by surajpaypay** [reply]
Thank you, this worked for me

**Reply by tishun** [reply]
Awesome, glad to help
